File: backend/src/main/java/de/htwg/tictactoe/TicTacToeModule.java
package de.htwg.tictactoe;

import com.google.inject.AbstractModule;
import com.google.inject.Singleton;

import de.htwg.tictactoe.controller.IController;
import de.htwg.tictactoe.controller.impl.Controller;
import de.htwg.tictactoe.model.IGame;
import de.htwg.tictactoe.model.impl.Game;


public class TicTacToeModule extends AbstractModule {

   @Override
   protected void configure() {

      bind(IGame.class).to(Game.class).in(Singleton.class);
      bind(IController.class).to(Controller.class).in(Singleton.class);
   }

}

File: backend/src/main/scala/tictactoe/TicTacToeModule.scala
package tictactoe

import com.google.inject.AbstractModule
import net.codingwell.scalaguice.ScalaModule
import tictactoe.persistence.{DevelopmentPersistenceEnvironment, PersistenceEnvironment}

/** Configuration for dependency injection with Guice
  *
  */
class TicTacToeModule extends AbstractModule with ScalaModule {

  def configure(): Unit = {
    bind[PersistenceEnvironment].toInstance(DevelopmentPersistenceEnvironment)
    bind[TicTacToeServer.type].toInstance(TicTacToeServer)
  }


}
